toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's Networks configuration schema.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+type Network struct {
+	BootProto   string   `json:"BootProto"`
+	GateWay     string   `json:"GateWay"`
+	Ip          string   `json:"Ip"`
+	NetMask     string   `json:"NetMask"`
+	OnBoot      bool     `json:"OnBoot"`
+	NameServers []string `json:"NameServer"`
+	Device      string   `json:"Device"`
+}
+
+// NetworkBootProto defines the different modes of network protocol
+type NetworkBootProto string
+
+const (
+	// NetworkBootProtoDHCP indicates that this is a DHCP protocol
+	NetworkBootProtoDHCP NetworkBootProto = "dhcp"
```

Comment:
Do these need to be public (start with a capital letter)? I think they are currently only used inside this file and only for the sake of listing valid protocols. If that's their only purpose, we could remove the constants completely and switch to a global list/map (see: my comment inside `bootProtoIsValid()`.

Reply:
Fixed